lib/ipmi_sol.c

```diff
@@ -1576,110 +1570,92 @@ ipmi_sol_red_pill(struct ipmi_intf * intf, int instance)
 				 * this down time when the connection is restarting, SOL will
 				 * exit even though the IPMC is available and the session is open.
 				 */
-				if (retrySol == MAX_SOL_RETRY)
-				{
+				if (retrySol == MAX_SOL_RETRY) {
 					/* no response to Get Device ID keepalive message */
 					bShouldExit = 1;
 					continue;
-				}
-				else
-				{
+				} else {
 					retrySol++;
 				}
-			}
-			else
-			{
+			} else {
 				/* if the keep Alive is successful reset retries to zero */
 				retrySol = 0;
 			}
 		} /* !oem="i82571spt" */
+
 		/* Wait up to half a second */
 		tv.tv_sec =  0;
 		tv.tv_usec = 500000;
 
 		retval = select(intf->fd + 1, &read_fds, NULL, NULL, &tv);
 
-		if (retval)
-		{
-			if (retval == -1)
-			{
+		if (retval) {
+			if (retval == -1) {
 				/* ERROR */
 				perror("select");
+				free(buffer);
```

Comment:
I wasn't sure if they needed to explicitly leave raw mode before exiting -- if so, this can be added here, or this can jump to a cleanup label.